File: 프로그래머스/2/42584. 주식가격/주식가격.java
class Solution {
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
        
        for(int i=0; i<prices.length;i++){
            for(int l=i+1;l<prices.length;l++){
            if(prices[i]>prices[l]||l==prices.length-1){
                answer[i]=l-i;
                break;
            }
        }
            
        }
        
        
        return answer;
    }
}

File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    n = len(prices)
    answer = [0] * n
    stack = []  # (index, price) 저장
    
    for i in range(n):
        # 현재 가격이 스택의 가격보다 작으면
        # 스택에 있는 가격들이 떨어진 것
        while stack and stack[-1][1] > prices[i]:
            idx, price = stack.pop()
            answer[idx] = i - idx
        
        stack.append((i, prices[i]))
    
    # 스택에 남은 것들은 끝까지 떨어지지 않은 것
    while stack:
        idx, price = stack.pop()
        answer[idx] = n - 1 - idx
    
    return answer